### DISCUSSION TITLE
Real-life project examples from A to Z (free or paid for)

### DISCUSSION THREAD

**Discussion by KayJay89** [explosion/spaCy#7617] [Help: Best practices]
Hi all!

First of all I want to express my gratitude towards the creators for making such a package open-source. With that said I would like to ask for some guidance in terms of how to approach this in a real-life instance.

I have read plenty pages of the documentation, experimented in parallel with going through them and also have followed each and every single of the online course. And while some concepts are definitely not completely understood, I have some of the (very) basics in my grey mass. What I can't seem to find are real (concise) real-life examples of how to use spaCy. 

But I am a person who struggles to make connections between the theory and some abstract examples and practice. I have searched endlessly on the big internet but I can't seem to find any real-life examples that guides newcomers like myself through the process of approaching a particular problem case. I understand the global idea behind how to train your model like it was explained in the video, but I fail to see how I can tackle this in practice myself. Mind you I have no computer science degree but I am home schooled in python and other languages. Also have no experience whatsoever with NLP or many of the concepts covered prior to starting the online course.

My current "problem" sounds very straightforward. Look into a PDF, find a certain string with particular characteristics, addresses, names,.. and put these into a database. While I can definitely get what I want through regex (at least in relation to the particular string type), I would love to be able to tackle that problem with spaCy and use this as a foundation to grow my knowledge. 

Basically what I am asking for is if there are any (free or paid) tutorials / examples of real-life applications that I could use to have some sort of idea of how to approach a particular problem. Even though I do realize there are probably hundreds of ways to go about the same issue.

I thank you in advance!
KR,
Kevin

**Comment by svlandeg** [comment]
Perhaps @koaning's tutorial series would be useful to you, to see a real-life case from conception to implementation? https://www.youtube.com/playlist?list=PLBmcuObd5An559HbDr_alBnwVsGq-7uTF

It won't cover your exact use-case, but it does explain the process from the first idea to a prototype, deals with data collection & evaluation, training an NER model from scratch, etc.

The original series was written for spaCy v2, but the latest episode introduces the migration to v3 as well.
